### PR TITLE
Add an example of slash command autocomplete

### PR DESCRIPTION
**Pull request by iomodo**
#### Summary
An example of slash command autocomplete with dynamic arguments. Just an example for testing purposes. No need of the review. Do not merge.

### CODE REVIEW

**Comment by hanzei** [comment]
@iomodo FYI: I tried to start the MM server will all three needed PRs, but the webapp crashes with:
```
index.ts:45 Uncaught (in promise) TypeError: Object(...) is not a function
    at Function.mapStateToProps [as mapToProps] (index.ts:45)
    at mapToPropsProxy (wrapMapToProps.js:41)
    at Function.detectFactoryAndVerify (wrapMapToProps.js:50)
    at mapToPropsProxy (wrapMapToProps.js:41)
    at handleFirstCall (selectorFactory.js:22)
    at pureFinalPropsSelector (selectorFactory.js:63)
    at Object.runComponentSelector [as run] (connectAdvanced.js:21)
    at Connect.initSelector (connectAdvanced.js:180)
    at new Connect(NeedsTeam) (connectAdvanced.js:118)
    at constructClassInstance (react-dom.development.js:11361)
```

**Comment by iomodo** [comment]
> @iomodo FYI: I tried to start the MM server will all three needed PRs, but the webapp crashes with:
> 
> ```
> index.ts:45 Uncaught (in promise) TypeError: Object(...) is not a function
>     at Function.mapStateToProps [as mapToProps] (index.ts:45)
>     at mapToPropsProxy (wrapMapToProps.js:41)
>     at Function.detectFactoryAndVerify (wrapMapToProps.js:50)
>     at mapToPropsProxy (wrapMapToProps.js:41)
>     at handleFirstCall (selectorFactory.js:22)
>     at pureFinalPropsSelector (selectorFactory.js:63)
>     at Object.runComponentSelector [as run] (connectAdvanced.js:21)
>     at Connect.initSelector (connectAdvanced.js:180)
>     at new Connect(NeedsTeam) (connectAdvanced.js:118)
>     at constructClassInstance (react-dom.development.js:11361)
> ```

@hanzei hmm strange, did you run `npm run dev` on redux?

**Comment by hanzei** [comment]
> @hanzei hmm strange, did you run `npm run dev` on redux?

Yes

File: server/command_hooks.go [hunk 1]
```diff
@@ -92,9 +92,84 @@ func (p *Plugin) registerCommands() error {
...
+
 	return nil
 }
 
+// CreateJiraAutocompleteData will create autocomplete data for jira plugin. For testing purposes only.
+func CreateJiraAutocompleteData() *model.AutocompleteData {
+	jira := model.NewAutocompleteData("jira", "[command]", "Available commands: connect, assign, disconnect, create, transition, view, subscribe, settings, install cloud/server, uninstall cloud/server, help")
+
+	connect := model.NewAutocompleteData("connect", "[server]", "Connect your Mattermost account to your Jira account")
+	connect.AddTextArgument("", "connect to the server", "[server]", "")
```
Comment (hanzei) [inline server/command_hooks.go:113]: Could you please add an example with a named argument?

**Comment by hanzei** [comment]
When I type /jira subscribe and press ENTER or TAB, the popup dismisses. I suppose this is a bug in the demo plugin.

From https://github.com/mattermost/mattermost-server/pull/14259#issuecomment-614642736

File: go.mod [hunk 2]
```diff
@@ -4,7 +4,7 @@ go 1.12
 
 require (
 	github.com/blang/semver v3.6.1+incompatible
-	github.com/mattermost/mattermost-server/v5 v5.3.2-0.20200129194125-99a82ef07ec0
-	github.com/pkg/errors v0.8.1
+	github.com/mattermost/mattermost-server/v5 v5.3.2-0.20200427133649-8f7f388eb27f
```
Comment (levb) [inline go.mod:7]: I am confused with the version requirement here, shouldn't a recent version be required here for Autocomplete?

Comment (iomodo) [reply go.mod:7]: @levb, now it points to the feature branch of a server. When we merge branch to the master I'll change this to point on the recent version.

**Comment by hanzei** [comment]
@iomodo Given that the server PR is closed, is this PR needed any longer?

**Comment by saturninoabril** [comment]
@hanzei @iomodo Can this PR reopen? I can see some benefits of this in terms of testing, both in manual and automated.

**Comment by hanzei** [comment]
I fully support the idea of adding an example for this feature, using Jira as an example might cause confusion. Hence, this PR has to be rewritten.

**Pull request closed without merging**